Build failure: chatterino2

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

`nix-build -A chatterino2`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/294721854

### Relevant log output

```console
[  3%] Building CXX object lib/libcommuni/src/util/CMakeFiles/Util.dir/irccommandparser.cpp.o
In file included from /nix/store/cp30fin83aimfljxb5988bkn2micgi04-qtbase-6.9.0/include/QtCore/qtmochelpers.h:21,
                 from /build/source/build/lib/libcommuni/src/util/Util_autogen/include/moc_irccommandparser.cpp:12,
                 from /build/source/lib/libcommuni/src/util/irccommandparser.cpp:602:
/nix/store/cp30fin83aimfljxb5988bkn2micgi04-qtbase-6.9.0/include/QtCore/q20algorithm.h:149:62: error: 'identity' in namespace 'Communi::q20' does not name a type
  149 |               typename Predicate, typename Projection = q20::identity>
      |                                                              ^~~~~~~~
/nix/store/cp30fin83aimfljxb5988bkn2micgi04-qtbase-6.9.0/include/QtCore/q20algorithm.h:162:62: error: 'identity' in namespace 'Communi::q20' does not name a type
  162 |               typename Predicate, typename Projection = q20::identity>
      |                                                              ^~~~~~~~
/nix/store/cp30fin83aimfljxb5988bkn2micgi04-qtbase-6.9.0/include/QtCore/q20algorithm.h:175:62: error: 'identity' in namespace 'Communi::q20' does not name a type
  175 |               typename Predicate, typename Projection = q20::identity>
      |                                                              ^~~~~~~~
make[2]: *** [lib/libcommuni/src/util/CMakeFiles/Util.dir/build.make:100: lib/libcommuni/src/util/CMakeFiles/Util.dir/irccommandparser.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:527: lib/libcommuni/src/util/CMakeFiles/Util.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[  3%] Building CXX object lib/settings/CMakeFiles/PajladaSettings.dir/src/settings/detail/rename.cpp.o
[  4%] Building CXX object lib/settings/CMakeFiles/PajladaSettings.dir/src/settings/detail/realpath.cpp.o
[  4%] Linking CXX static library libPajladaSettings.a
[  4%] Built target PajladaSettings
make: *** [Makefile:136: all] Error 2
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.14.2, NixOS, 25.05 (Warbler), 25.05.20250412.2631b0b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.28.1`
 - nixpkgs: `/nix/store/13ksj21mww6yshnhm8a419n89ycdwv2w-5ds20jm3x2s4z7wn3581r6lc9ybmh45b-source`

### Notify maintainers

@0Supa @NyCodeGHG 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ch/chatterino2/common.nix
+++ b/pkgs/by-name/ch/chatterino2/common.nix
@@ -7,6 +7,7 @@
   boost,
   openssl,
   libsecret,
+  libnotify,
   libavif,
   kdePackages,
 }:
@@ -24,13 +25,17 @@ stdenv.mkDerivation {
       qtsvg
       qt5compat
       qtkeychain
+      qtimageformats
     ])
     ++ [
       boost
       openssl
       libsecret
     ]
-    ++ lib.optional stdenv.hostPlatform.isLinux kdePackages.qtwayland
+    ++ lib.optionals stdenv.hostPlatform.isLinux [
+      kdePackages.qtwayland
+      libnotify
+    ]
     ++ lib.optional enableAvifSupport libavif;
 
   cmakeFlags = [

--- a/pkgs/by-name/ch/chatterino2/package.nix
+++ b/pkgs/by-name/ch/chatterino2/package.nix
@@ -12,13 +12,13 @@
   (
     finalAttrs: _: {
       pname = "chatterino2";
-      version = "2.5.2";
+      version = "2.5.3";
 
       src = fetchFromGitHub {
         owner = "Chatterino";
         repo = "chatterino2";
         tag = "v${finalAttrs.version}";
-        hash = "sha256-nrw4dQ7QjPPMbZXMC+p3VgUQKwc1ih6qS13D9+9oNuw=";
+        hash = "sha256-W2sqlqL6aa68aQ3nE161G64x7K7p8iByX03g1dseQbs=";
         fetchSubmodules = true;
       };
 

--- a/pkgs/by-name/ch/chatterino7/package.nix
+++ b/pkgs/by-name/ch/chatterino7/package.nix
@@ -13,13 +13,13 @@
   (
     finalAttrs: _: {
       pname = "chatterino7";
-      version = "7.5.2";
+      version = "7.5.3";
 
       src = fetchFromGitHub {
         owner = "SevenTV";
         repo = "chatterino7";
         tag = "v${finalAttrs.version}";
-        hash = "sha256-kQeW9Qa8NPs47xUlqggS4Df4fxIoknG8O5IBdOeIo+4=";
+        hash = "sha256-KrAr3DcQDjb+LP+vIf0qLSSgII0m5rNwhncLNHlLaC8=";
         fetchSubmodules = true;
       };
 

